common/src/main/java/com/google/auto/common/MoreElements.java

```diff
@@ -189,5 +195,24 @@ public static boolean isAnnotationPresent(Element element,
     return Optional.absent();
   }
 
+  /**
+   * Returns a {@link Predicate} that can be used to filter elements by {@link Modifier}.
+   * <p>
+   * Here is an example how one could get a List of static methods of a class:
+   * <code>
+   * FluentIterable.from(ElementFilter.methodsIn(clazzElement.getEnclosedElements()))
+   *     .filter(MoreElements.hasModifiers(Modifier.STATIC).toList();
+   * </code>
+   */
+  public static Predicate<Element> hasModifiers(Modifier... modifiers) {
```

Comment:
If you already have a `Set<Modifier>` or a `List<Modifier>` then it is messy to convert it into an array so you can use this method. That's because the `Collection<E>` is kind of crappy here, but we can't change that. I also think it would be OK to have both `(Modifier...)` and `(Set<Modifier>)` overloads.